pkg/generators/markers.go

```diff
@@ -305,25 +346,37 @@ func putNestedValue(m map[string]any, k []string, v any) (map[string]any, error)
 		var arrayDestination []any
 		if existing, ok := m[key]; !ok {
 			arrayDestination = make([]any, index+1)
-		} else {
+		} else if existing, ok := existing.([]any); !ok {
+			// Error case. Existing isn't of correct type. Can happen if
```

Comment:
I'm trying to understand this case better. Can you describe how it could be previously not an array? I'm just picturing assigning an item as both an array and not an array. Eg:

```
// +k8s:validation:cel="string message"
// +k8s:validation:cel[0]:rule="self.length() > 0"
```

Reply:
thats basically how